lib/AnalysisStructured/PtrAnalysis.cpp

```diff
@@ -297,6 +482,16 @@ LogicalResult PtrAnalysis::visitOperandAdd(arith::AddIOp addOp, PtrState &state,
     return failure();
   }
 
+  // When one state hasModulo while other state is not structured.
+  // Need to clear the modulo and use the operand as offset directly.
+  if (!lhsState.isStructured() && rhsState.hasModulo()) {
```

Comment:
does this mean modulo is not supported? can we support this?

Reply:
If lhs is not structured and rhs is structured but has modulo, my opinion is we still want to have support. We can do this in a separate PR to reduce complexity.

Reply:
Added TODO for this.